### PR TITLE
Add code coverage github action

### PR DESCRIPTION
**Pull request by joyet-simon**
### Reason for change

[Linear task](https://linear.app/almapay/issue/ECOM-1284/add-code-coverage-on-php-client)

We want to improve our quality code.

### Code changes

We add a github action to check the code coverage and block PR if it not respect our standards. 

### How to test

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [x] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [x] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: .github/workflows/code-coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+on:
+  push:
+    branches: [develop, master]
+  pull_request:
+    branches: [develop, master]
+  workflow_dispatch: ~
+
+permissions:
+  contents: write
```
Comment (gdraynz) [inline .github/workflows/code-coverage.yaml:11]: This means you intend on committing into the git repository in this workflow, I don't think that's necessary.

File: .github/workflows/code-coverage.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+jobs:
+  ci:
+    runs-on: ubuntu-20.04
```
Comment (gdraynz) [inline .github/workflows/code-coverage.yaml:15]:
```suggestion
    runs-on: ubuntu-22.04
```

File: .github/workflows/code-coverage.yaml (hunk 1 above)
Comment (joyet-simon) [reply .github/workflows/code-coverage.yaml:11]: I forgot to delete it

Comment (joyet-simon) [reply .github/workflows/code-coverage.yaml]: Done

File: .github/workflows/code-coverage.yaml (hunk 2 above)
Comment (joyet-simon) [reply .github/workflows/code-coverage.yaml]: Done

**Review by FranceBe — APPROVED** [review]

**Review by gdraynz — APPROVED** [review]

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

**Pull request merged**